src/gantry_calibration/calibration.py

```diff
@@ -0,0 +1,71 @@
+"""The core math and functionality behind the calibration algorithm."""
+
+import sys
+from collections import namedtuple
+import numpy as np
+
+
+Point = namedtuple("Point", ["x", "y"])
+PointPair = namedtuple("PointPair", ["expected", "actual"])
+
+
+class Calibration:
+    """Contains the calibration matrix and coefficients."""
+
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self, x_order: int, y_order: int, points: list[list[PointPair]]) -> None:
+        """Instantiate the object with expected and actual points."""
+        self.x_order = x_order
+        self.y_order = y_order
+        self.points = points
+        point_array = np.array(points)
+        self.x_count = point_array.shape[0]
+        self.y_count = point_array.shape[1]
+        if self.x_order >= self.x_count:
+            print(f"Error: x-axis order ({self.x_order}) must be less than count ({self.x_count})")
+            sys.exit()
```

Comment:
I don't think you should exit the program here. Just throw an exception containing the method. Sys.exit makes it very difficult to reuse this code. https://stackoverflow.com/questions/2052390/manually-raising-throwing-an-exception-in-python

Reply:
Refactored with your suggestions of raising exceptions.